chain/tipset.go

```diff
@@ -157,6 +163,32 @@ func (c *TipSetCache) Reset() {
 	c.len = 0
 }
 
+// Warm fills the TipSetCache with confidence tipsets so that subsequent calls to Add return a tipset.
+func (c *TipSetCache) Warm(ctx context.Context, head *types.TipSet, fn func(ctx context.Context, tsk types.TipSetKey) (*types.TipSet, error)) error {
```

Comment:
nit: A slightly better arg name?

```suggestion
func (c *TipSetCache) Warm(ctx context.Context, head *types.TipSet, tipsetGetter func(ctx context.Context, tsk types.TipSetKey) (*types.TipSet, error)) error {
```